### PR TITLE
update Google Apps Scripts types

### PR DESCRIPTION
**Pull request by dustinmichels**
@grant please review!

I ran the [dts-google-apps-script](https://github.com/motemen/dts-google-apps-script) type generator to produce these files. A number of files were updated and two new ones were added:

- google-apps-script.data-studio.d.ts
- google-apps-script.card-service.d.ts

### CODE REVIEW

File: types/google-apps-script/google-apps-script.drive.d.ts [hunk 1]
```diff
@@ -32,104 +32,29 @@ declare namespace GoogleAppsScript {
     export interface DriveApp {
       Access: typeof Access;
       Permission: typeof Permission;
-      /**
-       * Adds the given file to the root of the user's Drive.
-       * This method does not move the file out of its existing parent folder;
-       * a file can have more than one parent simultaneously.
-       */
```
Comment (grant) [inline types/google-apps-script/google-apps-script.drive.d.ts]: Don't remove these.

**Review by grant — CHANGES_REQUESTED** [review]
Please:
- Do not delete any interfaces or comments. They may need to be hand-added.
- Use `string`, not `String.

Looks fine otherwise. Thanks!

File: types/google-apps-script/google-apps-script.drive.d.ts (hunk 1 above)
Comment (dustinmichels) [reply types/google-apps-script/google-apps-script.drive.d.ts]: Thanks for reviewing! I did not change anything in the files auto-generated by [dts-google-apps-script](https://github.com/motemen/dts-google-apps-script), so I will be manually replacing `String` with `string` and re-inserting comments, just to be clear.

Comment (dustinmichels) [reply types/google-apps-script/google-apps-script.drive.d.ts]: Also, I'm not sure why the Travis CI tests are failing or how to make them pass.

**Comment by dustinmichels** [comment]
@grant I believe I have successfully added back comments/ interfaces, and changed all occurrences of `String` to `string`. Let me know what you think!

File: types/google-apps-script/google-apps-script.gmail.d.ts [hunk 2]
```diff
@@ -207,12 +208,13 @@ declare namespace GoogleAppsScript {
      */
     export interface GmailMessage {
       createDraftReply(body: string): GmailDraft;
-      createDraftReply(body: string, options: GmailDraftOptions): GmailDraft;
```
Comment (grant) [inline types/google-apps-script/google-apps-script.gmail.d.ts]: Please undo `Object` -> `GmailDraftOptions`.
Keep `GmailDraftOptions`.

**Comment by grant** [comment]
Looks good. Unfortunately the generator is a bit broken, so manual changes have to be made.

Can you fix the `GmailDraftOptions` part? Then we can merge.

**Review by grant — APPROVED** [review]

**Comment by dustinmichels** [comment]
Deleted!

**Comment by grant** [comment]
@dustinmichels FYI, I don't think this will be merged until Travis passes.

**Comment by dustinmichels** [comment]
Hmm okay. Any idea what I can do to make the tests pass? I can't really tell what's wrong.

**Comment by dustinmichels** [comment]
It looks like there could be an issue with the `google-apps-script.card.d.ts` file I deleted.

```
{ Error: ENOENT: no such file or directory, open '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/google-apps-script/google-apps-script.card.d.ts'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/google-apps-script/google-apps-script.card.d.ts' }
```

Is that file being referenced by something else?

**Comment by dustinmichels** [comment]
I fixed an the error with `google-apps-script.card.d.ts` being in the index, and another one with `google-apps-script.data-studio.d.ts` _not_ being in the index.

Now there's an error with `OverGridImage` not being found in `google-adwords-scripts`, which is a dependency of this script?

Line 2289 of the Travis message:
```
google-adwords-scripts failing:
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/google-apps-script/google-apps-script.spreadsheet.d.ts:1386:20
ERROR: 1386:20   expect  TypeScript@next compile error: 
Cannot find name 'OverGridImage'.
```

**Comment by dustinmichels** [comment]
Is this because there is no `export interface OverGridImage` in the `spreadsheet.d.ts` file? Does this need to be created manually?

**Comment by grant** [comment]
I can't help interpret the errors you see. Follow the @typescript-bot instructions.

For example, the logs show this error:
```
Error: Unused file /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/google-apps-script/google-apps-script.card-service.d.ts
```

**Comment by dustinmichels** [comment]
I believe I fixed that error, and a few others. The[ most recent failure](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/454586123?utm_source=github_status&utm_medium=notification) though has to do with the "google-adwords-scripts" failing (because it `Cannot find name 'OverGridImage'.`) which I'm not sure how to fix.

It seems like google-adwords-scripts is a dependency of this package. I'm not sure how an object like `OverGridImage` (which is new in this pull request) is supposed to be defined so a dependency like google-adwords-scripts can find it, but it seems this wasn't done automatically by the generator script.

I'm sorry for bombarding you with so many messages! I really want to be helpful and get the new types updated, but there are a lot of pieces I don't really understand. I really appreciate your patience & helpfulness.

**Comment by grant** [comment]
I think for the `OverGridImage` error, you'll need to see what the error is in VSCode for `types/google-apps-script/google-apps-script.spreadsheet.d.ts`.

Perhaps smaller PRs would be merged faster.

**Comment by dustinmichels** [comment]
@grant thanks for the feedback. I'll work on a smaller PR that doesn't fail tests.

**Pull request closed without merging**